src/main.ts

```diff
@@ -154,9 +154,25 @@ export function generateFile(ctx: Context, fileDesc: FileDescriptorProto): [stri
     ) {
       chunks.push(makeProtobufTimestampWrapper());
     }
+
+    if (
+      fileDesc.messageType.find((message) =>
+        message.field.find((field) => field.typeName === ".google.protobuf.Struct")
+      )
+    ) {
+      const structFieldNames = {
+        nullValue: maybeSnakeToCamel("null_value", ctx.options),
+        numberValue: maybeSnakeToCamel("number_value", ctx.options),
+        stringValue: maybeSnakeToCamel("string_value", ctx.options),
+        boolValue: maybeSnakeToCamel("bool_value", ctx.options),
+        structValue: maybeSnakeToCamel("struct_value", ctx.options),
+        listValue: maybeSnakeToCamel("list_value", ctx.options),
+      };
+      chunks.push(makeProtobufStructWrapper(options, structFieldNames));
+    }
   }
 
-  if (options.outputEncodeMethods || options.outputJsonMethods || options.outputTypeRegistry) {
+  if (options.outputEncodeMethods || options.outputJsonMethods || options.outputTypeRegistry || options.nestJs) {
```

Comment:
Suggestion: Ah okay, we add the `|| options.nestJs` b/c we need it to generate any wrap/unwrap methods...

I wonder if we could make this a little more nuanced, by doing something like:

```ts
const wrap = this.generateWrap(...);
const unwrap = this.generateUnwrap(...);
if (options.outputEncodeMethods || ... || wrap.length > 0 || unwrap.length > 0)
```

Pretty minor, but I think it'd avoid outputting some of the empty `export const Hero = {}` lines that are unnecessary.

Reply:
I'll look into that

Reply:
We can't seem to do that as props of `generateWrap()` and `generateUnwrap()` are in the scope of the callback of `visit` inside that `if`

Any other suggestion to clean up the output?

Reply:
Ah shoot, you're right. I had to poke around a little bit, but I think pushed up a few more conditionals that cleaned up some of the unnecessary output.